core/unroller.cpp

```diff
@@ -123,7 +125,31 @@ UnorderedTermMap & Unroller::var_cache_at_time(unsigned int k)
     }
   }
 
-  return time_cache_[k];
+  UnorderedTermMap & var_cache = time_cache_[k];
+  size_t current_num_vars = ts_.statevars().size();
+  current_num_vars += ts_.inputvars().size();
+
```

Comment:
Actually, one minor thing -- let's put a comment here explaining what this check is for. Just want to unroll any recently added variables.